File: app/src/main/java/com/globalwarming/earthsaver/group/GroupUserAdapter.java
package com.globalwarming.earthsaver.group;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.globalwarming.earthsaver.R;
import com.globalwarming.earthsaver.User;
import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.ArrayList;
import java.util.List;

public class GroupUserAdapter extends RecyclerView.Adapter<GroupUserAdapter.ViewHolder> {

    private CollectionReference db;
    private List<String> users = new ArrayList<>();

    public GroupUserAdapter() {
        db = FirebaseFirestore.getInstance().collection("users");
    }

    public void setList(List<String> users) {
        this.users.clear();
        this.users.addAll(users);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_user, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        String userId = users.get(position);
        db.document(userId).get().addOnCompleteListener(task -> {
            if (task.isSuccessful()) {
                User user = task.getResult().toObject(User.class);
                String points = "Points : 0";
                if (user.getPoints() != null) {
                    points = "Points : " + user.getPoints();
                }
                holder.textViewName.setText(user.getName());
                holder.textViewPoints.setText(points);
                holder.textViewAddress.setText(user.getLocation());
            }
        });
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        TextView textViewName;
        TextView textViewPoints;
        TextView textViewAddress;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewName = itemView.findViewById(R.id.text_view_name);
            textViewPoints = itemView.findViewById(R.id.text_view_points);
            textViewAddress = itemView.findViewById(R.id.text_view_address);
        }

    }

}

File: app/src/main/java/com/globalwarming/earthsaver/group/GroupUserAdapter.kt
package com.globalwarming.earthsaver.group

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.globalwarming.earthsaver.R
import com.globalwarming.earthsaver.User
import com.globalwarming.earthsaver.account.ProfileActivity
import com.globalwarming.earthsaver.databinding.LayoutUserBinding
import com.google.firebase.firestore.FirebaseFirestore
import java.util.*

class GroupUserAdapter(
    private val group: Group
) : RecyclerView.Adapter<GroupUserAdapter.ViewHolder>() {

    private val db = FirebaseFirestore.getInstance().collection("users")
    private val users = ArrayList<String>()

    fun setList(users: List<String>) {
        this.users.clear()
        this.users.addAll(users)
        notifyDataSetChanged()
    }

    fun removeUser(userId: String) {
        val index = users.indexOfFirst {
            it.contains(userId)
        }
        if (index > -1) {
            users.removeAt(index)
            notifyItemRemoved(index)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = LayoutUserBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val userId = users[position].split("|")[1]
        db.document(userId).get().addOnCompleteListener { task ->
            val user = task.result?.toObject(User::class.java)
            if (user != null) {
                holder.binding.textViewName.text = user.name
                holder.binding.textViewPoints.text = "Points : ${user.points}"
                holder.binding.textViewAddress.text = user.location
                Glide.with(holder.binding.imageView).load(user.image)
                    .placeholder(R.drawable.logo)
                    .error(R.drawable.logo)
                    .into(holder.binding.imageView)
            }
        }

        holder.itemView.setOnClickListener {
            val intent = Intent(it.context, ProfileActivity::class.java)
            intent.putExtra(ProfileActivity.USER_ID, userId)
            intent.putExtra(ProfileActivity.GROUP, group)
            it.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return users.size
    }

    class ViewHolder(val binding: LayoutUserBinding) : RecyclerView.ViewHolder(binding.root)

}